util/xrh_header.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+func ParseXRHIDHeader(inputIdentity string) (XRHIdentity *identity.XRHID, err error) {
```

Comment:
I would suggest using local variables instead of named return values: https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values

Reply:
I agree with this - my initial thought of this was that the `XRHIdentity` was at the package scope - it was really confusing.

Reply:
I agree, named params removed